### PR TITLE
ORC-1304: [C++] Fix bug of seeking over empty PRESENT stream

### PR DESCRIPTION
**Pull request by coderex2522**
### What changes were proposed in this pull request?
This pr doesn't call readHeader function when the seek location is all zero.

### Why are the changes needed?
When the length of present stream is zero, the seek function pre-call readHeader() function will throw ParseError. 

### How was this patch tested?
Add the ByteRle.seekOverEmptyPresentStream and  RLEv1.seekOverEmptyPresentStream UT.

### CODE REVIEW

**Review by wgtmac — COMMENTED** [review]
We need to add e2e test cases to cover at least empty struct/map/list mentioned in the issue.

**Comment by stiga-huang** [comment]
+1. The fix LGTM. I suggest a more specific commit title, e.g. "ORC-1304: [C++] Fix bug of seeking in empty PRESENT stream".

For the missing e2e tests, we can add them in TestPredicatePushdown.cc.

**Review by dongjoon-hyun — COMMENTED** [review]
+1 for @wgtmac 's and @stiga-huang 's comment.

**Comment by coderex2522** [comment]
> We need to add e2e test cases to cover at least empty struct/map/list mentioned in the issue.

I add the new UT WriterTest.testSeekWithNestedType to cover the child column of the struct/map/list type has empty present stream.

File: c++/test/TestPredicatePushdown.cc [hunk 1]
```diff
@@ -557,4 +557,103 @@ namespace orc {
       TestFirstStripeSelectedWithStripeStats(reader.get(), pos);
     }
   }
+
+  TEST(TestPredicatePushdown, testSeekOverEmptyPresentStream) {
```
Comment (wgtmac) [inline c++/test/TestPredicatePushdown.cc:561]: nit: the bug has nothing to do with predicate pushdown, we'd better move it out of here.

File: c++/test/TestWriter.cc [hunk 2]
```diff
@@ -2044,103 +2044,5 @@ namespace orc {
     testSuppressPresentStream(CompressionKind_SNAPPY);
   }
 
-  TEST(WriterTest, testSeekWithNestedType) {
```
Comment (wgtmac) [inline c++/test/TestWriter.cc]: Why is it removed?

File: c++/test/TestPredicatePushdown.cc (hunk 1 above)
Comment (coderex2522) [reply c++/test/TestPredicatePushdown.cc]: I move the ut into the TestReader.cc file because the ParseError exception was not handled during the read operation.

File: c++/test/TestWriter.cc (hunk 2 above)
Comment (coderex2522) [reply c++/test/TestWriter.cc]: It just move into TestReadIntent.testSeekOverEmptyPresentStream in the TestReader.cc file.

**Review by wgtmac — APPROVED** [review]
+1 LGTM.

**Comment by dongjoon-hyun** [comment]
cc @williamhyun , too
